### DISCUSSION TITLE
Integrate semgrep in CI on every PR as well as CI deep

### DISCUSSION THREAD

**Discussion by smol-ninja** [sablier-labs/evm-monorepo#1037] [Ideas]
### Leaving some links here
- [Running semgrep on lockup `main` branch](https://app.warp.dev/block/QfFqqUDS0iVbg0Iwyrltea)
- [Official website](https://semgrep.dev/)
- [Custom rules for smart contract](https://github.com/Decurity/semgrep-smart-contracts/tree/master?tab=readme-ov-file)
- [Uniswap v4 CI using semgrep](https://github.com/Uniswap/v4-core/actions/runs/10210213833/workflow)

### Benefits
- Up to date rules based on new public findings in solidity code
- Automatic checks against latest vulnerabilities in Sablier codebase through CI
- Custom rules allow filtering redundant and outdated disclosures such as using `unchecked` for `++i` in loop

### Running semgrep on `v1.0.0`

<img width="728" alt="Screenshot 2024-09-05 at 23 46 57" src="https://github.com/user-attachments/assets/ee6fe978-8176-4d4c-8c4f-d40b2d69a35f">

cc @sablier-labs/solidity

**Comment by PaulRBerg** [comment]
Seems interesting. Can't see any issue in experimenting with this.
